hudi-flink-datasource/hudi-flink1.18.x/src/main/java/org/apache/hudi/table/format/cow/ParquetSplitReaderUtil.java

```diff
@@ -546,15 +606,30 @@ private static WritableColumnVector createWritableColumnVector(
           // schema evolution: read the file with a new extended field name.
           int fieldIndex = getFieldIndexInPhysicalType(rowType.getFields().get(i).getName(), groupType);
           if (fieldIndex < 0) {
-            columnVectors[i] = (WritableColumnVector) createVectorFromConstant(rowType.getTypeAt(i), null, batchSize);
+            if (groupType.getRepetition().equals(Type.Repetition.REPEATED) && !rowType.getTypeAt(i).is(LogicalTypeRoot.ARRAY)) {
```

Comment:
I'm not sure this will work in all cases. Since there isn't a descriptor, that can't be used to check the repetition level. I did add a unit test for adding a new field to an Array of rows and Map of rows.